src/isolate.rs

```diff
@@ -254,34 +266,156 @@ impl Isolate {
     Ok(())
   }
 
+  pub fn mod_new(
+    &mut self,
+    name: String,
+    source: String,
+  ) -> Result<libdeno::deno_mod, JSError> {
+    let name_ = CString::new(name.clone()).unwrap();
+    let name_ptr = name_.as_ptr() as *const i8;
+
+    let source_ = CString::new(source.clone()).unwrap();
+    let source_ptr = source_.as_ptr() as *const i8;
+
+    let id = unsafe {
+      libdeno::deno_mod_new(self.libdeno_isolate, name_ptr, source_ptr)
+    };
+    if let Some(js_error) = self.last_exception() {
+      assert_eq!(id, 0);
+      return Err(js_error);
+    }
+
+    let name2 = name.clone();
+    self.modules.insert(id, ModuleInfo { name });
+
+    debug!("modules_by_name insert {}", name2);
+    self.modules_by_name.insert(name2, id);
+
+    Ok(id)
+  }
+
+  // TODO(ry) This should be private...
+  pub fn resolve_cb(
+    &self,
+    specifier: &str,
+    referrer: libdeno::deno_mod,
+  ) -> libdeno::deno_mod {
+    self
+      .state
+      .metrics
+      .resolve_count
+      .fetch_add(1, Ordering::SeqCst);
```

Comment:
Ordering::Relaxed seems strict enough here. SeqCst is definitely overkill -- if you want to play safe use Ordering::AckRel .

Reply:
Fixed.